static/js/components/JobDetails.jsx

```diff
@@ -1,13 +1,33 @@
 var AddForm = React.createClass({
 
   getInitialState: function() {
-    return {clicked: false, errored: false};
+    return {clicked: false, errored: false, JSON_workload_checked: false};
+  },
+
+  raiseJSONWorkloadWarning: function(workload) {
+    if (workload.length === 0 || workload[0] != "{") {
```

Comment:
Do you think preceding spaces will be a common use case? Now that I think of it, is there any reason not to strip leading spaces in general?